src/MoBi.Presentation/Presenter/EditModuleConfigurationsPresenter.cs

```diff
@@ -0,0 +1,203 @@
+      {
+         _view.AddModuleConfigurationNode(_treeNodeFactory.CreateFor(moduleConfiguration));
+      }
+
+      public void RemoveModuleConfiguration(ITreeNode selectedTreeNode)
+      {
+         if (selectedTreeNode == null)
+            return;
+
+         var moduleConfigurationToRemove = moduleConfigurationDTOFor(selectedTreeNode);
+         _moduleConfigurations.Remove(moduleConfigurationToRemove);
+         _view.RemoveNodeFromSelectedView(selectedTreeNode);
+         addModuleToSelectionView(moduleConfigurationFor(selectedTreeNode).Module);
+      }
+
+      public void SelectedModuleConfigurationNodeChanged(ITreeNode selectedTreeNode)
+      {
+         var dto = moduleConfigurationDTOFor(selectedTreeNode);
+         _view.UnbindModuleConfiguration();
+
+         // if the user selects a non-module node we want disable the binding
+         if (dto == null)
+         {
+            _view.EnableRemove = false;
+            return;
+         }
+
+         _view.EnableRemove = true;
+         _view.BindModuleConfiguration(dto);
+      }
+
+      public void SelectedModuleNodeChanged(ITreeNode selectedNode)
+      {
+         var module = projectModuleFor(selectedNode);
+
+         _view.EnableAdd = module != null;
+      }
+
+      private Module projectModuleFor(ITreeNode selectedNode)
+      {
+         return selectedNode?.TagAsObject as Module;
+      }
+
+      private ModuleConfiguration moduleConfigurationFor(ITreeNode treeNode)
+      {
+         return treeNode?.TagAsObject as ModuleConfiguration;
+      }
+      
+      private ModuleConfigurationDTO moduleConfigurationDTOFor(ITreeNode treeNode)
+      {
+         var moduleConfiguration = moduleConfigurationFor(treeNode);
+         return dtoFor(moduleConfiguration);
+      }
+
+      public IReadOnlyList<MoleculeStartValuesBuildingBlock> MoleculeStartValuesCollectionFor(ITreeNode selectedNode)
+      {
+         if (selectedNode == null)
+            return new List<MoleculeStartValuesBuildingBlock>();
+
+         var dto = moduleConfigurationDTOFor(selectedNode);
+         return dto.MoleculeStartValuesCollection;
+      }
+
+      public IReadOnlyList<ParameterStartValuesBuildingBlock> ParameterStartValuesCollectionFor(ITreeNode selectedNode)
+      {
+         if (selectedNode == null)
+            return new List<ParameterStartValuesBuildingBlock>();
+
+         var dto = moduleConfigurationDTOFor(selectedNode);
+         return dto.ParameterStartValuesCollection;
+      }
+
+      public int CompareSelectedNodes(ITreeNode node1, ITreeNode node2)
+      {
+         var dto1 = moduleConfigurationDTOFor(node1);
+         var dto2 = moduleConfigurationDTOFor(node2);
+
+         return _moduleConfigurations.IndexOf(dto1) - _moduleConfigurations.IndexOf(dto2);
+      }
+
+      public IReadOnlyList<ModuleConfiguration> ModuleConfigurations => _moduleConfigurations.Select(x => x.ModuleConfiguration).ToList();
+
+      public bool CanDrag(ITreeNode node)
+      {
```

Comment:
any reason why line 160 is on one line and lie 162 not? I'd suggest to always shorten one liner in one line 

public bool CanDrag(ITreeNode node)=>nodeIsModuleConfiguration(node);